web/react/components/sidebar.jsx

```diff
@@ -157,6 +125,7 @@ export default class Sidebar extends React.Component {
             privateChannels,
             visibleDirectChannels,
             hiddenDirectChannelCount,
+            unreadCounts: JSON.parse(JSON.stringify(ChannelStore.getUnreadCounts())),
```

Comment:
Yeah, I think that's probably better and I think that anytime the store returns anything it should be a deep copy but I think I'd like to talk about this in the developer meeting first. Right now I'd like to leave it as is just so the stores are consistent. 0/5 though, if you feel strongly about it I can move it to the store